internal/interfaces/interfaces.go

```diff
@@ -74,3 +76,37 @@ func ImplementsError(t gotypes.Type) bool {
 func ImplementsGomegaMatcher(t gotypes.Type) bool {
 	return t != nil && gotypes.Implements(t, gomegaMatcherType)
 }
+
+// Note: we cannot check for an argument which _implements_ gomega.Gomega without adding a Go
+// module dependency on gomega.  This is because the gomega.Gomega interface definition references
+// gomega.AsyncAssertion, whose methods return gomega.AsyncAssertion. Because Go does not have
+// interface covariance or contravariance, any "local copy" of gomega.AsyncAssertion cannot
+// be satisified by any actual `gomega.AsyncAssertion` implementation, as the methods do not return
+// local.AsyncAssertion but rather gomega.AsyncAssertion.
+//
+// Also, Gomega probably doesn't even accept an argument whose type implements the interface, but
+// rather whose type _is_ the interface. So this check should suffice.
+func IsGomega(t gotypes.Type) bool {
+	named, isNamed := t.(*gotypes.Named)
+	if !isNamed {
+		return false
+	}
+
+	obj := named.Obj()
+
+	if obj.Name() != "Gomega" {
+		return false
+	}
+
+	return isPackageSymbol(named.Obj(), "github.com/onsi/gomega/types", "Gomega")
+}
+
+func isPackageSymbol(obj types.Object, pkgPath, name string) bool {
+	if obj.Name() != name {
+		return false
+	}
+
+	vendorPieces := strings.Split(pkgPath, "/vendor/")
+
+	return pkgPath == vendorPieces[len(vendorPieces)-1]
+}
```

Comment:
I think, but not sure, that we can use the `gomegahandler` package for this check. The handler only exists if the gomega package is imported, and holds the knowledge if it's dot import or named import.

So you'll only need to check if the type name is `Gomega` and the left side, if exists, is expected by the handler. I think it simpler. WDYT?